File: C#/0594. Longest Harmonious Subsequence.cs
public class Solution {
    public int FindLHS(int[] nums) {
        if(nums.Length<=1){
            return 0;
        }
        Array.Sort(nums);
        int left=0;
        int target=nums[left];
        int rep=0;
        for(int i=1;i<nums.Length;i++){
            if(nums[i]-target>1){
                if (nums[i-1]-target==1){
                    rep=Math.Max(rep,i-left);
                }
                while(nums[left]==target){
                    left++;
                }
                target=nums[left];
            }
        }
        if (nums[nums.Length-1]-target==1){
            rep=Math.Max(rep,nums.Length-left);
        }
        
        return rep;
        
    }
}

File: 20190519python/0594. Longest Harmonious Subsequence.py
class Solution:
    def findLHS(self, nums: List[int]) -> int:
        dict_nums={}
        for num in nums:
            if num not in dict_nums:
                dict_nums[num]=1
            else:
                dict_nums[num]+=1
                
        
        nums=list(set(nums))
        nums.sort()
        
        rep=0
        for k in range(len(nums)-1):
            if nums[k+1]==nums[k]+1:
                rep=max(rep,dict_nums[nums[k]]+dict_nums[nums[k+1]])
        
        return rep
